### PR TITLE
Add `embargo_status` to dandiset endpoint

### PR DESCRIPTION
**Pull request by dchiquito**
Added in https://github.com/dandi/dandi-api/pull/643

@jwodder should I merge this before or after I merge the change in dandi-api?

### CODE REVIEW

File: dandi/tests/test_dandiapi.py [hunk 1]
```diff
@@ -515,6 +515,7 @@ def test_remote_dandiset_json_dict(text_dandiset):
         "created": anys.ANY_AWARE_DATETIME_STR,
         "modified": anys.ANY_AWARE_DATETIME_STR,
         "contact_person": anys.ANY_STR,
+        "embargo_status": anys.ANY_STR,
```
Comment (jwodder) [inline dandi/tests/test_dandiapi.py:518]: This isn't enough.  You also need to actually add the field definition to `RemoteDandiset` in `dandiapi.py`, in the manner of, e.g., [`contact_person`](https://github.com/dandi/dandi-cli/blob/b825018c8a2d36bf1f1b516a4959e753e979dcf0/dandi/dandiapi.py#L680-L683).

**Review by jwodder — CHANGES_REQUESTED** [review]
This isn't enough.  You also need to actually add the field definition to `RemoteDandiset` in `dandiapi.py`, in the manner of, e.g., [`contact_person`](https://github.com/dandi/dandi-cli/blob/b825018c8a2d36bf1f1b516a4959e753e979dcf0/dandi/dandiapi.py#L680-L683).

**Comment by jwodder** [comment]
@dchiquito 

> should I merge this before or after I merge the change in dandi-api?

Merge dandi-api first so that we can properly test dandi-cli.

**Comment by yarikoptic** [comment]
@dchiquito please have a look to fix up linting and let's see current state of tests

**Comment by dchiquito** [comment]
@jwodder I'm not sure why the windows/mac tests would pass while ubuntu fails. Are there any fundamental differences between them?

**Comment by jwodder** [comment]
@dchiquito The Ubuntu tests are the only ones that include the tests of the Dockerized API.

**Comment by dchiquito** [comment]
@jwodder what triggers https://hub.docker.com/r/dandiarchive/dandiarchive-api to be updated? Perusing https://github.com/dandi/dandi-api/blob/master/.github/workflows/cli-integration.yml suggests pushes to master, but I'm not clear if that includes merges or not.
Once the docker hub is updated I believe these tests should pass.

**Comment by jwodder** [comment]
@dchiquito The Docker Hub image should be updated whenever the master branch of the repo is updated, whether by a merge or a direct push.

**Review by jwodder — APPROVED** [review]

**Pull request merged**